src/CommandLineUtils/Internal/CommandLineProcessor.cs

```diff
@@ -533,9 +535,10 @@ public CommandArgumentEnumerator(IEnumerator<CommandArgument> enumerator)
 
             public bool MoveNext()
             {
-                if (Current == null || !Current.MultipleValues)
+                if (!_currentValid || !Current.MultipleValues)
```

Comment:
It's also better because it not only checks that `MoveNext` has been called, but also that it returned true. The error was caused by calling `Current` after `MoveNext` has returned false.